fission/src/systems/preferences/PreferencesSystem.ts

```diff
@@ -136,6 +151,34 @@ class PreferencesSystem {
         return allFieldPrefs
     }
 
+    /**
+     * @param {string} miraName - The name of the motor assembly to get preference for.
+     * @returns {MotorPreferences} Motor preferences found for the given motor, or default motor preferences if none are found.
+     */
+    public static getMotorPreferences(miraName: string): MotorPreferences {
+        const allMotorPrefs = this.getAllMotorPreferences()
+
+        if (allMotorPrefs[miraName] == undefined) {
+            const defaultPrefs = DefaultMotorPreferences(miraName)
+            allMotorPrefs[miraName] = defaultPrefs
+            return defaultPrefs
+        }
+
+        return allMotorPrefs[miraName]
+    }
+
+    /** @returns Preferences for every motor that was found in local storage. */
+    public static getAllMotorPreferences(): { [key: string]: MotorPreferences } {
+        let allMotorPrefs = this.getPreference<{ [key: string]: MotorPreferences }>(MotorPreferencesKey)
+
+        if (allMotorPrefs == undefined) {
+            allMotorPrefs = {}
+            this._preferences[MotorPreferencesKey] = allMotorPrefs
+        }
+
+        return allMotorPrefs
```

Comment:
It seems like `this._preferences[MotorPreferencesKey]` should equal `allMoterPrefs` no matter what, so we can remove this `if` statement and just have an unnecessary assignment some of the time. In my opinion, this is cleaner and doesn't actually incur any performance penalty because branching is slow.

```suggestion
        return (this._preferences[MotorPreferencesKey] =
            this.getPreference<{ [key: string]: MotorPreferences }>(MotorPreferencesKey) ?? {})
```